packages/bot-web-ui/src/components/dashboard/dashboard-component/cards.tsx

```diff
@@ -1,64 +1,160 @@
+    setActiveTab,
+    setFileLoaded,
+    showVideoDialog,
+}: TCardProps) => {
+    type TCardArray = {
+        icon: string;
+        content: string;
+        method: () => void;
+        disable: string | '';
+    };
+    const openGoogleDriveDialog = () => {
+        showVideoDialog({
+            type: 'google',
+            component: <GoogleDrive />,
+            url: '',
+        });
+    };
+    const file_input_ref = React.useRef<HTMLInputElement | null>(null);
+    const [is_file_supported, setIsFileSupported] = React.useState<boolean>(true);
+
+    const clear_preview_ref = React.useRef<HTMLInputElement | null>(null);
+    const openFileLoader = () => {
+        file_input_ref?.current?.click();
+    };
+
+    const actions: TCardArray[] = [
         {
             icon: 'IcMyComputer',
-            label: localize('My computer'),
+            content: localize('My computer'),
+            method: openFileLoader,
+            disable: has_file_loaded ? 'tab__dashboard__table__disabled-card' : '',
         },
         {
             icon: 'IcGoogleDriveDbot',
-            label: localize('Google Drive'),
-            method: onDriveConnect,
+            content: localize('Google Drive'),
+            method: openGoogleDriveDialog,
+            disable: has_file_loaded ? 'tab__dashboard__table__disabled-card' : '',
         },
         {
             icon: 'IcBotBuilder',
-            label: localize('Bot Builder'),
+            content: localize('Bot Builder'),
             method: () => setActiveTab(1),
+            disable: '',
         },
         {
             icon: 'IcQuickStrategy',
-            label: localize('Quick Strategy'),
+            content: localize('Quick Strategy'),
             method: () => setActiveTab(2),
+            disable: '',
         },
     ];
 
     return React.useMemo(
         () => (
-            <div className='dc-tabs__dashboard-cards-wrapper' id='db-dashboard-cards'>
-                {actions.map(({ icon, label, method }) => (
-                    <div key={label} className='dc-tabs__dashboard-card'>
-                        <Icon
-                            className='dc-tabs__dashboard-card__image'
-                            size='8rem'
-                            icon={icon}
-                            id={icon}
-                            onClick={method}
-                        />
-                        <Text size='xs' line_height='s' color='prominent'>
-                            {label}
-                        </Text>
-                    </div>
-                ))}
+            <div className='tab__dashboard__table'>
+                <div className='tab__dashboard__table__tiles' id='tab__dashboard__table__tiles'>
+                    {actions.map((icons, index) => {
+                        const { icon, content, method, disable } = icons;
+                        return (
+                            <div key={index} className={`${disable} tab__dashboard__table__block`}>
```

Comment:
```suggestion
                            <div key={content} className={`${disable} tab__dashboard__table__block`}>
```
Best to use a unique key instead of `index`

Reply:
done ✅